### ISSUE TITLE
push version tags to dockerhub 

### ISSUE THREAD

**Issue by pred2k** [spikecodes/libreddit#188]
## Is your feature request related to a problem? Please describe.

Current latest Image (ID 7b4323c3b25a) fail to run
```
docker run -it -p 8080:8080 spikecodes/libreddit
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
standard_init_linux.go:219: exec user process caused: exec format error
```

## Describe the solution you'd like
Please also consider adding a image tag with the version to 
`docker build -t libreddit:v0.10.1 -t libreddit:latest .`

This allows a easy rollback to an older version, pinning to a version (for whatever reason) and identify the libreddit version easily inside the docker image.

**Comment by pred2k** [comment]
Looks like its this format in github Action:
```
      - name: Build and push
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile.arm64
          platforms: linux/arm64
          push: true
          tags: spikecodes/libreddit:latest,spikecodes/libreddit:v0.10.1
```
https://github.com/docker/build-push-action#inputs

**Comment by Nachtalb** [comment]
I haven't worked with `docker/build-push-action@v2` yet but from what I see is that due to using 2 different dockerfiles you will have to push at least two tags one with arm one with amd etc. 
That means the main one will either be arm or amd like eg. `latest` which supports `amd64` (and `i386`) and `latest-arm` which supports `arm64` (and other arm versions as  `arm/v7` & `arm/v6`)

Something like this:
```yml
      - name: Build and push AMD64
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile
          platforms: linux/amd64,linux/i386
          push: true
          tags: spikecodes/libreddit:latest

      - name: Build and push ARM64
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile.arm64
          platforms: linux/arm64,linux/arm/v7,linux/arm/v6
          push: true
          tags: spikecodes/libreddit:latest-arm

```

Alternatively of course, if you manage to merge the Dockerfiles into one you could just use `platforms: linux/amd64,linux/arm64` etc. under the same `latest` tag which then would support both platforms. 

ATM everyone gets the arm build even if he has a amd64 or i386 host and gets the error mentioned by OP `standard_init_linux.go:219: exec user process caused: exec format error` 


As you tag your work you can put in the a docker tag with version like so:
```yml
      - id: get_version
        uses: battila7/get-version-action@v2
      - name: Build and push AMD64
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile
          platforms: linux/amd64,linux/i386
          push: true
          tags: spikecodes/libreddit:latest, spikecodes/libreddit:${{ steps.get_version.outputs.version }}

```

**Comment by spikecodes** [comment]
Thank for this issue! I've actually moved the compiling of `amd64` builds to Docker Hub which makes them compile in only 12 minutes (as opposed to one hour). Though Docker Hub only supports that architecture so `arm64` still builds over GitHub Actions (that's why the workflow is now "Docker ARM Build").

That error from Docker image 7b4323c3b25a is because I was trying to push two architectures to the same tag accidentally. I've just pushed 9f1ba27 which will now publish `arm64` builds to the `spikecodes/libreddit:arm` tag and `amd64` builds to the `spikecodes/libreddit:latest` tag.

The new tag has been documented in the README for future Docker selfhosters! Let me know if there are any other issues.
